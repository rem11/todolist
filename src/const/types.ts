/**
 * Params for the screens from root navigatior
 */
export type RootStackParamList = {
  Lock: undefined;
  'TODO List': undefined;
};
