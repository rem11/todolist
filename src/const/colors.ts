export const COLOR_WHITE = '#fff';
export const COLOR_BLACK = '#000';
