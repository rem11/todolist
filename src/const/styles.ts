import {StyleSheet} from 'react-native';

export const commonStyles = StyleSheet.create({
  text: {
    fontSize: 16,
    lineHeight: 24,
  },
});
