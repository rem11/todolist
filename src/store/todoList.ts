import {PayloadAction, createSlice} from '@reduxjs/toolkit';
import {TodoListState, TodoRecord} from '../const/types';

const initialState: TodoListState = {
  todoList: [],
  latestId: 0,
};

const todoList = createSlice({
  name: 'todoList',
  initialState,
  reducers: {
    addRecord: (state, action: PayloadAction<string>) => {
      state.todoList.push({
        id: ++state.latestId,
        text: action.payload,
      });
    },
    removeRecord: (state, action: PayloadAction<number>) => {
      const index = state.todoList.findIndex(
        record => record.id === action.payload,
      );
      if (index !== -1) {
        state.todoList.splice(index, 1);
      }
    },
    updateRecord: (state, action: PayloadAction<TodoRecord>) => {
      const {id} = action.payload;
      const index = state.todoList.findIndex(record => record.id === id);
      if (index !== -1) {
        state.todoList[index] = action.payload;
      }
    },
  },
});

export const {addRecord, removeRecord, updateRecord} = todoList.actions;

export default todoList.reducer;
