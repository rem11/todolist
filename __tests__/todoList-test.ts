import {TodoListState} from '../src/const/types';
import todoList, {
  addRecord,
  removeRecord,
  updateRecord,
} from '../src/store/todoList';

const testInitialState: TodoListState = {
  todoList: [
    {
      id: 1,
      text: 'Record 1',
    },
    {
      id: 2,
      text: 'Record 2',
    },
    {
      id: 3,
      text: 'Record 3',
    },
  ],
  latestId: 3,
};

test('test add record', () => {
  expect(todoList(testInitialState, addRecord('Record 4'))).toMatchSnapshot();
});

test('test remove record', () => {
  expect(todoList(testInitialState, removeRecord(2))).toMatchSnapshot();
});

test('test update record', () => {
  expect(
    todoList(
      testInitialState,
      updateRecord({id: 2, text: 'Record 2 (updated)'}),
    ),
  ).toMatchSnapshot();
});
