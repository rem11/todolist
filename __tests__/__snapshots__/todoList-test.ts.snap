// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test add record 1`] = `
{
  "latestId": 4,
  "todoList": [
    {
      "id": 1,
      "text": "Record 1",
    },
    {
      "id": 2,
      "text": "Record 2",
    },
    {
      "id": 3,
      "text": "Record 3",
    },
    {
      "id": 4,
      "text": "Record 4",
    },
  ],
}
`;

exports[`test remove record 1`] = `
{
  "latestId": 3,
  "todoList": [
    {
      "id": 1,
      "text": "Record 1",
    },
    {
      "id": 3,
      "text": "Record 3",
    },
  ],
}
`;

exports[`test update record 1`] = `
{
  "latestId": 3,
  "todoList": [
    {
      "id": 1,
      "text": "Record 1",
    },
    {
      "id": 2,
      "text": "Record 2 (updated)",
    },
    {
      "id": 3,
      "text": "Record 3",
    },
  ],
}
`;
